### PR TITLE
Correct unsubscribe logic for subscriptions

### PR DESCRIPTION
**Pull request by clone1018**
Ensures that when subscriptions are unsubscribed from a user, that we keep their benefits until the end of the billing cycle. This will ensure you cannot spam sub / unsub and get charged a bunch of money, and makes sure subscribers get their full benefits.﻿

### CODE REVIEW

File: lib/glimesh/formatters.ex [hunk 1]
```diff
@@ -8,17 +8,16 @@ defmodule Glimesh.Formatters do
...
   def format_datetime(timestamp) when is_integer(timestamp) do
-    d = DateTime.from_unix!(timestamp)
-    "#{d.year}/#{d.month}/#{d.day}"
+    format_datetime(DateTime.from_unix!(timestamp))
   end
 
-  def format_datetime(%{year: year, month: month, day: day}) do
-    "#{year}/#{month}/#{day}"
+  def format_datetime(%{year: _, month: _, day: _y} = datetime) do
+    Calendar.strftime(datetime, "%b %d %Y")
```
Comment (clone1018) [inline lib/glimesh/formatters.ex:16]: March 2, 2021

File: lib/glimesh/payment_providers/stripe/stripe.ex [hunk 2]
```diff
@@ -174,11 +174,12 @@ defmodule Glimesh.PaymentProviders.StripeProvider do
     potential_payout_amount = total_amount - glimesh_cut
 
     # Withholding is calculated from the potential payout amount, not the full amount
+    # Rounded UP in the event the withholding is a fractional percent eg 2.50 - 0.125 = 2.38
     withholding_amount =
       if streamer.tax_withholding_percent,
         do:
           Decimal.to_integer(
-            Decimal.mult(potential_payout_amount, streamer.tax_withholding_percent)
+            Decimal.round(Decimal.mult(potential_payout_amount, streamer.tax_withholding_percent))
```
Comment (clone1018) [inline lib/glimesh/payment_providers/stripe/stripe.ex:182]: Fixes a bug where the withholding percent could add fractional pennies, should always round up.

File: lib/glimesh/payments.ex [hunk 3]
```diff
@@ -207,16 +207,66 @@ defmodule Glimesh.Payments do
...
+           }) do
+      {:ok, sub}
+    else
+      {:error, %Stripe.Error{} = error} -> {:error, error.user_message || error.message}
+      {:error, %Ecto.Changeset{errors: errors}} -> {:error, errors}
+    end
+  end
+
+  def resubscribe(%Subscription{is_active: false}) do
+    {:error, "Cannot resubscribe a expired subscription."}
```
Comment (mja00) [inline lib/glimesh/payments.ex:233]:
```suggestion
    {:error, "Cannot resubscribe an expired subscription."}
```

**Review by mja00 — APPROVED** [review]
Just one grammar thing

**Pull request merged**